### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by Andrew77788**
- [DEMO LINK](https://Andrew77788.github.io/layout_calendar/)
- [TEST REPORT LINK](https://Andrew77788.github.io/layout_calendar/report/html_report/)

### CODE REVIEW

File: src/styles/main.scss [hunk 1]
```diff
@@ -1,3 +1,66 @@
...
+    @for $day from 1 through 31 {
+      &:nth-child(#{$day})::before {
+        content: '#{$day}';
+      }
+    }
+
+  }
+
+  @each $start-day, $first-cell in $days-week {
+    &_start-day-#{$start-day}
```
Comment (etojeDenys) [inline src/styles/main.scss]:
```suggestion
    &-start-day-#{$start-day}
```

File: src/styles/main.scss [hunk 2]
```diff
@@ -1,3 +1,66 @@
...
+      :first-child {
+        margin-left: calc(#{$first-cell} * #{$cell-size + 1px});
+      }
+  }
+
+  @for $last-day from 28 to 31 {
+    &_length-#{$last-day}
```
Comment (etojeDenys) [inline src/styles/main.scss]:
```suggestion
    &-length-#{$last-day}
```

File: src/index.html [hunk 3]
```diff
@@ -1,12 +1,45 @@
...
   <head>
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <title>Calendar</title>
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
-  <body>
-    <h1>Calendar</h1>
+
+  <body class="calendar calendar---start-day-mon calendar---length-31">
```
Comment (etojeDenys) [inline src/index.html]:
```suggestion
  <body class="calendar calendar-start-day-mon calendar-length-31">
```

File: src/styles/main.scss [hunk 4]
```diff
@@ -1,3 +1,66 @@
...
+$animation-duration: 0.5s;
+$days-week: (
+  'mon':0,
+  'tue':1,
+  'wed':2,
+  'thu':3,
+  'fri':4,
+  'sat':5,
+  'sun':6,
+);
```
Comment (etojeDenys) [inline src/styles/main.scss]: move vars to separate file

**Review by etojeDenys — CHANGES_REQUESTED** [review]

File: src/index.html (hunk 3 above)
Comment (Viktor-Kost) [inline src/index.html]: either use as mentor showed or do like this (to follow BEM recommendations)
```suggestion
  <body class="calendar calendar--start-day--mon calendar--month-length--31">
```

File: src/styles/calendar.scss [hunk 5]
```diff
@@ -0,0 +1,50 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  padding: 10px;
+  width: 706;
```
Comment (Viktor-Kost) [inline src/styles/calendar.scss]: this value shouldn't be hardcoded. use calculation based on cell width and gap between them

File: src/styles/calendar.scss [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  font-size: 30px;
+  font-family: Arial, Helvetica, sans-serif;
```
Comment (Viktor-Kost) [inline src/styles/calendar.scss]: move to `html` tag

File: src/styles/calendar.scss [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+  &__day {
+    width: #{$cell-size};
+    height: #{$cell-size};
+
+    background-color: $background-color;
+    border: 1px solid $border-color;
+
+    box-sizing: border-box;
```
Comment (Viktor-Kost) [inline src/styles/calendar.scss]: move to `*` selector

File: src/styles/calendar.scss [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    &:hover {
+      cursor: pointer;
+      background-color: $hover-color;
+      transform: translateY(-20px);
+      transition-duration: $animation-duration;
```
Comment (Viktor-Kost) [inline src/styles/calendar.scss:28]: move to `&__day` and add `transition-property` as well

File: src/styles/calendar.scss [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    @for $day from 1 through 31 {
+      &:nth-child(#{$day})::before {
+        content: '#{$day}';
+      }
+    }
+
+  }
+
+  @each $start-day, $first-cell in $days-week {
+    &---start-day-#{$start-day}
```
Comment (Viktor-Kost) [inline src/styles/calendar.scss]:
```suggestion
    &--start-day--#{$start-day}
```

File: src/styles/calendar.scss [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+      :first-child {
+        margin-left: calc(#{$first-cell} * #{$cell-size + 1px});
+      }
+  }
+
+  @for $last-day from 28 to 31 {
+    &---length-#{$last-day}
```
Comment (Viktor-Kost) [inline src/styles/calendar.scss]:
```suggestion
    &--length--#{$last-day}
```

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]
Great job!
Some minor fixes needed

File: src/styles/calendar.scss (hunk 7 above)
Comment (Andrew77788) [reply src/styles/calendar.scss]: Не розумію, ось цей момент, якщо я зроблю так 
*{   background-color: $background-color;
border: 1px solid $border-color;
box-sizing: border-box;
}
це ж не буде правильно, що ви мали на увазі ?)

**Review by OleziO — APPROVED** [review]

**Pull request closed without merging**